collector/generic_collector.go

```diff
@@ -83,12 +87,18 @@ func NewCollector(collectorName string, configFile []byte) (*GenericCollector, e
 		minPollFrequency = minSupportedFrequency
 	}
 
+	if len(configInJSON.MetricsConfig) > metricCountLimit {
+		return nil, fmt.Errorf("Too many metrics defined: %d limit: %d", len(configInJSON.MetricsConfig), metricCountLimit)
```

Comment:
This is a collector for user defined metrics. There are two of them 
* generic - that can consume arbitrary metric via a regexp
* prometheus - that requires that the metrics are in prometheus format